code/Test_definitions/DeviceIdentifier_retrieve_identifier.feature

```diff
@@ -0,0 +1,141 @@
+
+  @DeviceIdentifier_retrieve_identifier101_success
+  Scenario:  retrieve device identifer for phone number PHONENUMBER2, network connection and access token matches PHONENUMBER2
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And one of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When the HTTPS "POST" request is sent
+    And the request body has the field phoneNumber with a value of PHONENUMBER2
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response header "Content-Type" is "application/json"
+    And the response body complies with the OAS schema at "/components/schemas/200RetrieveIdentifier"
+    Then the response status code is 200
+    And the response property "$.imeisv" is IMEISV2
+
+  @DeviceIdentifier_retrieve_identifier103_same_device_different_SIMs
+  Scenario:  retrieve device identifer on the same device but for different SIMs by different telcos. The device identifier MUST be the same.
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And one of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When a device_identifier1 is retrieved on the device with SIMCARD1 in slot1
+    And a device_identifier2 is retrieved on the same device with SIMCARD2 in slot1
+    Then device_identifier1 and device_identifier2 are equal
+
+
+  @DeviceIdentifier_retrieve_identifier102_ipv4Address_success
+  Scenario:  retrieve device identifer for phone number PHONENUMBER2, network connection and access token matches PHONENUMBER2
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And one of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When the HTTPS "POST" request is sent
+    And the request body has the field ipv4Address.publicAddress with a value of PUBLICIPV4ADDRESS
+    And the request body has the field ipv4Address.publicPort with a value of PUBLICPORT
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response header "Content-Type" is "application/json"
+    And the response body complies with the OAS schema at "/components/schemas/200RetrieveIdentifier"
+    Then the response status code is 200
+    And the response property "$.imeisv" is IMEISV2
+
+  @DeviceIdentifier_retrieve_identifier200_missing_device_information
+  Scenario:  retrieve identifier but no device information in request
+    Given they use the base url
+    And the resource is "/retrieve-identifier"
+    And one of the scopes associated with the access token is device-identifier:retrieve-identifier
+    When the HTTPS "POST" request is sent
+    And the connection the request is sent over originates from a device with PHONENUMBER1
+    And the request body has NO the field phoneNumber nor other device information
```

Comment:
The OAS definition does not allow empty request bodies to be sent. It only allow no request body or a populated request body to be sent
```suggestion
    And there is no request body
```